lib/delta-sync/file-sync.js

```diff
@@ -0,0 +1,1051 @@
+ *
+ * @async
+ * @private
+ * @function queryFullFiles
+ * @return {Array} Array with the bindings of the query
+ */
+async function queryFullFiles() {
+  const queryString = `
+    ${PREFIXES}
+
+    SELECT ?vuuid ?puri
+    WHERE {
+      GRAPH <${TEMP_FILE_GRAPH}> {
+        ?vuri a nfo:FileDataObject              ;
+              a ext:DownloadSuccess             ;
+              mu:uuid               ?vuuid      ;
+              nfo:fileName          ?filename   ;
+              dct:format            ?format     ;
+              nfo:fileSize          ?filesize   ;
+              dbpedia:fileExtension ?extension  ;
+              dct:created           ?created    ;
+              dct:modified          ?modified   .
+        ?puri a nfo:FileDataObject              ;
+              mu:uuid               ?puuid      ;
+              nie:dataSource        ?vuri       ;
+              nfo:fileName          ?pfilename  ;
+              dct:format            ?pformat    ;
+              nfo:fileSize          ?pfilesize  ;
+              dbpedia:fileExtension ?pextension ;
+              dct:created           ?pcreated   ;
+              dct:modified          ?pmodified  .
+        FILTER NOT EXISTS { ?vuri a ext:RemapSuccess .
+                            ?vuri a ext:RemapFailure . }
+      }
+    }
+  `;
+  const results = await querySudo(queryString);
+  return results.results.bindings;
+}
+
+/**
+ * Produce and execute a query on the database in the temporary removal graph to ask for the filenames of physical files. This only selects URIs for files where the metadata is complete, and takes care to deal with replacements that came from remapping.
+ *
+ * @async
+ * @private
+ * @function queryForPFilenames
+ * @return {Array} Array of path strings to the physical files that need to be removed.
+ */
+async function queryForPFilenames() {
+  try {
+    let queryString = `
+      ${PREFIXES}
+
+      SELECT ?new2puri
+      WHERE {
+        GRAPH <${TEMP_FILE_REMOVAL_GRAPH}> {
+          ?vuri a nfo:FileDataObject              ;
+                mu:uuid               ?vuuid      ;
+                nfo:fileName          ?filename   ;
+                dct:format            ?format     ;
+                nfo:fileSize          ?filesize   ;
+                dbpedia:fileExtension ?extension  ;
+                dct:created           ?created    ;
+                dct:modified          ?modified   .
+          ?puri a nfo:FileDataObject              ;
+                mu:uuid               ?puuid      ;
+                nie:dataSource        ?vuri       ;
+                nfo:fileName          ?pfilename  ;
+                dct:format            ?pformat    ;
+                nfo:fileSize          ?pfilesize  ;
+                dbpedia:fileExtension ?pextension ;
+                dct:created           ?pcreated   ;
+                dct:modified          ?pmodified  .
+        }
+        GRAPH <${INGEST_GRAPH}> {
```

Comment:
I don't really understand what you mean here.
Files need to be fully published first (because metadata is only published when all triples have arrived) before they can be removed. If you remove a file first, the insert wins and the removal had no effect. It all depends on timing, but usually, there is enough time between insert and removal of the same file to not cause a problem.